### PR TITLE
Move maps & sets indexed by GlobRef.t into the kernel

### PR DESCRIPTION
**Pull request by vbgl**
This moves the modules `RefOrdered`, `RefOrdered_env`, `Refmap`, `Refmap_env`, `Refset`, and `Refset_env` from `Globnames` to `Names`.

### CODE REVIEW

File: kernel/names.ml
Comment (ejgallego) [inline kernel/names.ml]: I think current trend is to use `GlobRef.Map` etc...

File: library/globnames.ml
Comment (ejgallego) [inline library/globnames.ml]: This needs a deprecation notice.

**Review by ejgallego — APPROVED** [review]
Modulo the module naming, this looks good to me.

File: kernel/names.ml
Comment (ppedrot) [reply kernel/names.ml]: Note that for constants and inductives, we still use the old scheme. The reason is probably that there are two different notions of maps, one based on canonical names and the other on user names.

**Review by amahboubi — APPROVED** [review]
The change in the code of the ring tactic is innocuous. But it's the only part I've checked.

**Comment by vbgl** [comment]
I’ve just pushed a patch with the renaming suggested by Emilio. I was a bit annoyed by all these `t` types. I don’t know if there is a way to achieve the same effects without introducing yet an other name for the type t:

```
module M = struct
  type t = A
  type alias = t
  module Inner = struct
    type t = alias
  end
end
```

**Comment by SkySkimmer** [comment]
`type nonrec t = t`

**Comment by ejgallego** [comment]
> I’ve just pushed a patch with the renaming suggested by Emilio. I was a bit annoyed by all these t types. I don’t know if there is a way to achieve the same effects without introducing yet an other name for the type t:

Indeed, apart from the solution of @SkySkimmer , `include` can help sometimes, and having a signature for Ordered may allow a slightly cleaner style as done for `Id.Map` in the `mli`.

**Review by ppedrot — APPROVED** [review]

**Comment by maximedenes** [comment]
@vbgl do you want to squash the commits? Aren't they fine as they are?

**Review by maximedenes — APPROVED** [review]

**Comment by vbgl** [comment]
Yes. I will squash them (some changes introduced in the first one are reverted in the second one).

I thought that this PR was waiting for approval from a few more people: this is why i waited for squashing.

**Comment by maximedenes** [comment]
You can squash before they review, so that CI runs meanwhile. But anyway for this kind of PRs (one real change and some propagation), I think the merging documentation says you can proceed with approval only by the maintainers impacted by the change, not the propagation.

**Review by herbelin — APPROVED** [review]
The move seems good. Regarding the recommended naming policy, I'm trusting the advice of @ejgallego who has a good overview of the current state of the API.

**Pull request merged**
